flink-runtime/src/main/java/org/apache/flink/runtime/externalresource/ExternalResourceUtils.java

```diff
@@ -0,0 +1,213 @@
+			} else {
+				externalResourceAmountMap.put(resourceName, amountOpt.get());
+			}
+		}
+
+		return externalResourceAmountMap;
+	}
+
+	/**
+	 * Instantiate the {@link ExternalResourceDriver ExternalResourceDrivers} for all of enabled external resources. {@link ExternalResourceDriver ExternalResourceDrivers}
+	 * are mapped by its resource name.
+	 */
+	@VisibleForTesting
+	static Map<String, ExternalResourceDriver> externalResourceDriversFromConfig(Configuration config, PluginManager pluginManager) {
+		final Set<String> resourceSet = getExternalResourceSet(config);
+		LOG.info("Enabled external resources: {}", resourceSet);
+
+		if (resourceSet.isEmpty()) {
+			return Collections.emptyMap();
+		}
+
+		final Iterator<ExternalResourceDriverFactory> factoryIterator = pluginManager.load(ExternalResourceDriverFactory.class);
+		final Map<String, ExternalResourceDriverFactory> externalResourceFactories = new HashMap<>();
+		factoryIterator.forEachRemaining(
+			externalResourceDriverFactory -> externalResourceFactories.put(externalResourceDriverFactory.getClass().getName(), externalResourceDriverFactory));
+
+		final Map<String, ExternalResourceDriver> externalResourceDrivers = new HashMap<>();
+		for (String resourceName: resourceSet) {
+			final ConfigOption<String> driverClassOption =
+				key(ExternalResourceOptions.keyWithResourceNameAndSuffix(resourceName, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX))
+					.stringType()
+					.noDefaultValue();
+			final String driverFactoryClassName = config.getString(driverClassOption);
+			if (StringUtils.isNullOrWhitespaceOnly(driverFactoryClassName)) {
+				LOG.warn("Could not find driver class name for {}. Please make sure {} is configured.",
+					resourceName, driverClassOption.key());
+				continue;
+			}
+
+			ExternalResourceDriverFactory externalResourceDriverFactory = externalResourceFactories.get(driverFactoryClassName);
+			if (externalResourceDriverFactory != null) {
+				DelegatingConfiguration delegatingConfiguration =
+					new DelegatingConfiguration(config, ExternalResourceOptions.keyWithResourceNameAndSuffix(resourceName, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_PARAM_SUFFIX));
+				try {
+					externalResourceDrivers.put(resourceName, externalResourceDriverFactory.createExternalResourceDriver(delegatingConfiguration));
+					LOG.info("Add external resources driver for {}.", resourceName);
+				} catch (Exception e) {
+					LOG.warn("Could not instantiate driver with factory {} for {}. {}", driverFactoryClassName, resourceName, e);
+				}
+			} else {
+				LOG.warn("Could not find factory class {} for {}.", driverFactoryClassName, resourceName);
+			}
+		}
+
+		return externalResourceDrivers;
+	}
+
+	/**
+	 * Instantiate the {@link ExternalResourceInfoProvider} for all of enabled external resources.
+	 */
+	public static ExternalResourceInfoProvider getExternalResourceInfoProvider(Configuration config, PluginManager pluginManager) {
```

Comment:
If we rename the `ExternalResourceInfoProvider` implementation into `StaticExternalResourceInfoProvider` I would suggest to rename this method into `createStaticExternalResourceInfoProvider`.